setup/generate_setup_file.py

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env python3
+
+from typing import IO, List
+
+import os
+import sys
+import traceback
+
+
+class SetupGenerator:
+    def __init__(self, tue_env_dir=None):
+        if tue_env_dir is None:
+            tue_env_dir = os.environ["TUE_ENV_DIR"]
```

Comment:
raise exception if this is still None

Reply:
I think this will raise a `KeyError` when not existing. When the user sets it to an empty string. Than it is useless, but the script still works.